### PR TITLE
Feature/oidc

### PR DESCRIPTION
**Pull request by reavdhoef**
Added funtionality for accepting OIDC protocol for authentication.

### CODE REVIEW

File: controllers/User.php [hunk 1]
```diff
@@ -8,50 +8,112 @@
...
+	];
+
+	$context = stream_context_create($options);
+	$result = file_get_contents($tokenUrl, false, $context);
+	$jsonresult = json_decode($result, TRUE);
+
+	$claimElement = explode('.', $jsonresult['id_token'])[1];
+	$claimData = json_decode(base64_decode($claimElement), TRUE);
+
+        $this->session->unset_userdata('username');
```
Comment (lwesterhof) [inline controllers/User.php:44]: Whitespace

File: views/user/callback.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+HOI
```
Comment (lwesterhof) [inline views/user/callback.php]: Leftover debug statements?

**Review by lwesterhof — CHANGES_REQUESTED** [review]

File: controllers/User.php (hunk 1 above)
Comment (reavdhoef) [reply controllers/User.php:44]: Redid the entire file, replacing tabs by spaces and realigned everything again.

**Review by lwesterhof — APPROVED** [review]
LGTM

**Pull request merged**